src/components/movies-list/movies-list.jsx

```diff
@@ -1,16 +1,35 @@
-import React from 'react';
+import React, {useState, useEffect, useMemo, useCallback} from 'react';
 import PropTypes from 'prop-types';
 
 import SmallMovieCard from './small-movie-card';
+import ShowMoreButton from '../show-more-button/show-more-button';
+
+const PAGE_LIMIT = 8;
 
 const MoviesList = ({movies}) => {
-  return (
+  const [limit, setLimit] = useState(PAGE_LIMIT);
+
+  useEffect(() => {
+    setLimit(PAGE_LIMIT);
+  }, [movies]);
+
+  const handleShowMoreButtonClick = useCallback(() => {
+    setLimit((currentLimit) => currentLimit + PAGE_LIMIT);
+  }, [setLimit]);
+
+  const renderedMovies = useMemo(() => {
+    return Array.from({length: Math.min(limit, movies.length)}, (_, index) => {
+      const movie = movies[index];
+      return <SmallMovieCard key={movie.id} movie={movie}/>;
+    });
```

Comment:
Как вариант попроще:
```suggestion
    return movies.slice(0, limit).map((movie) => {
      return <SmallMovieCard key={movie.id} movie={movie}/>;
    });
```